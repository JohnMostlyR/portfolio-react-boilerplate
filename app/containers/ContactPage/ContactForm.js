import React from 'react';
import PropTypes from 'prop-types';
import isLength from 'validator/lib/isLength';
import isEmail from 'validator/lib/isEmail';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';

import messages from './messages';
import validateForm from './ValidateForm';

import Form, {
  Footer,
  InfoList,
  InfoListItem,
  Input,
  SubmitButton,
  TextArea,
} from '../../components/Form';

function ContactForm({
  fieldError,
  field,
  changeHandler,
  intl,
  onSubmitHandler,
  sendStatus,
}) {
  const { subject, message, name, email } = field;
  let formIsInvalid = validateForm({ field, fieldError });

  function handleKeyDown() {
    formIsInvalid = validateForm({ field, fieldError });
  }

  return (
    <Form onSubmit={onSubmitHandler} onChange={handleKeyDown}>
      <div>
        <div>
          <div>
            <InfoList>
              <InfoListItem>
                <FormattedMessage {...messages.requirementOne} />
              </InfoListItem>
            </InfoList>
          </div>
          <div>
            <Input
              helperText={intl.formatMessage(messages.subjectText)}
              label={intl.formatMessage(messages.subjectLabel)}
              maxLength={50}
              minLength={3}
              name="subject"
              changeHandler={changeHandler}
              validate={val =>
                isLength(val, { min: 3, max: 50 })
                  ? false
                  : intl.formatMessage(messages.subjectError)
              }
              value={subject}
            />
            <TextArea
              helperText={intl.formatMessage(messages.yourMessageText)}
              label={intl.formatMessage(messages.yourMessageLabel)}
              maxLength={300}
              minLength={5}
              name="message"
              changeHandler={changeHandler}
              validate={val =>
                isLength(val, { min: 5, max: 300 })
                  ? false
                  : intl.formatMessage(messages.yourMessageError)
              }
              value={message}
            />
            <Input
              helperText={intl.formatMessage(messages.nameText)}
              label={intl.formatMessage(messages.nameLabel)}
              maxLength={50}
              minLength={2}
              name="name"
              placeholder={intl.formatMessage(messages.namePlaceholder)}
              changeHandler={changeHandler}
              validate={val =>
                isLength(val, { min: 2, max: 50 })
                  ? false
                  : intl.formatMessage(messages.nameError)
              }
              value={name}
            />
            <Input
              helperText={intl.formatMessage(messages.emailText)}
              inputType="email"
              label={intl.formatMessage(messages.emailLabel)}
              name="email"
              placeholder={intl.formatMessage(messages.emailPlaceholder)}
              changeHandler={changeHandler}
              validate={val =>
                isEmail(val) ? false : intl.formatMessage(messages.emailError)
              }
              value={email}
            />
          </div>
          <Footer>
            <SubmitButton
              buttonState={sendStatus}
              formIsValid={!formIsInvalid}
            />
          </Footer>
        </div>
      </div>
    </Form>
  );
}

ContactForm.propTypes = {
  field: PropTypes.object,
  fieldError: PropTypes.object,
  changeHandler: PropTypes.func.isRequired,
  intl: intlShape,
  onSubmitHandler: PropTypes.func.isRequired,
  sendStatus: PropTypes.oneOf(['idle', 'sending', 'success', 'error']),
};

export default injectIntl(ContactForm);
