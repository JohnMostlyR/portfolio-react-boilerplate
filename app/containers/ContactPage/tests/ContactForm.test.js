import React from 'react';
import toJson from 'enzyme-to-json';

import ContactForm from '../ContactForm';
// import validateForm from '../ValidateForm';
// import Form from '../../../components/Form';

import { shallowWithIntl } from '../../../helpers/intl-enzyme-test-helper';

jest.mock('../ValidateForm');

describe('<ContactForm />', () => {
  it('should render and match the snapshot', () => {
    const changeHandler = jest.fn();
    const onSubmitHandler = jest.fn();
    const wrapper = shallowWithIntl(
      <ContactForm
        changeHandler={changeHandler}
        onSubmitHandler={onSubmitHandler}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it.skip('should call "changeHandler" on a Change event', () => {
    // TODO: Finish this test
    const changeHandler = jest.fn();
    const onSubmitHandler = jest.fn();
    const wrapper = shallowWithIntl(
      <ContactForm
        changeHandler={changeHandler}
        onSubmitHandler={onSubmitHandler}
      />,
    );

    // console.log(wrapper.debug());
    // console.log(wrapper.dive().find(Form));
    wrapper.simulate('onChange', { target: { value: true } });
    // expect(validateForm).toHaveBeenCalledTimes(1);
  });
});
