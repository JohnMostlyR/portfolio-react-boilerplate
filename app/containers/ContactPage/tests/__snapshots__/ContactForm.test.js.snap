// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm /> should render and match the snapshot 1`] = `
<ContactForm
  changeHandler={[Function]}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "portfolio.component.BrandLink.title": "← Back to the homepage",
        "portfolio.component.ContentLoadingIndicator.error": "Terribly sorry, but the content cannot be loaded at this time",
        "portfolio.component.ContentLoadingIndicator.pastDelay": "Loading content",
        "portfolio.component.ContentLoadingIndicator.timedOut": "Loading is taking longer than expected",
        "portfolio.components.Form.helperText.range": "Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}",
        "portfolio.components.Form.helperText.valid": "Thanks!",
        "portfolio.components.LanguageMenu.buttonLabel": "Language",
        "portfolio.components.LanguageMenu.label": "Choose a Language",
        "portfolio.components.LanguageMenu.title": "Language menu",
        "portfolio.components.SubmitButton.error": "Failed. Please try again",
        "portfolio.components.SubmitButton.idle": "Send",
        "portfolio.components.SubmitButton.sending": "Sending...",
        "portfolio.components.SubmitButton.success": "Sent! I'll respond as soon as possible",
        "portfolio.containers.ContactPage.emailError": "Your email address is required.",
        "portfolio.containers.ContactPage.emailLabel": "Your email address",
        "portfolio.containers.ContactPage.emailPlaceholder": "(e.g. my@email.com)",
        "portfolio.containers.ContactPage.emailText": "Please add your email address here.",
        "portfolio.containers.ContactPage.header": "CONTACT ME",
        "portfolio.containers.ContactPage.nameError": "Your name is required.",
        "portfolio.containers.ContactPage.nameLabel": "Your name",
        "portfolio.containers.ContactPage.namePlaceholder": "(e.g. John Doo)",
        "portfolio.containers.ContactPage.nameText": "Please add your name here.",
        "portfolio.containers.ContactPage.requirementOne": "Please fill in all fields",
        "portfolio.containers.ContactPage.subjectError": "A subject is required.",
        "portfolio.containers.ContactPage.subjectLabel": "Subject",
        "portfolio.containers.ContactPage.subjectText": "Please provide a subject for your message.",
        "portfolio.containers.ContactPage.yourMessageError": "A message is required.",
        "portfolio.containers.ContactPage.yourMessageLabel": "Your message",
        "portfolio.containers.ContactPage.yourMessageText": "Please add your message here.",
        "portfolio.nav.about": "About",
        "portfolio.nav.contact": "Contact",
        "portfolio.nav.header": "Site navigation",
        "portfolio.nav.home": "Intro",
        "portfolio.nav.projects": "Projects",
        "portfolio.nav.showMenuButton": "MENU",
        "portfolio.nav.skills": "Skills",
        "portfolio.page.AboutPage.pageDescription": "A short description about me.",
        "portfolio.page.AboutPage.pageTitle": "About me : Johan Meester's portfolio",
        "portfolio.page.AboutPage.title": "ABOUT ME",
        "portfolio.page.NotFoundPage.content": "I am sorry, but the page you requested does not exist or has been removed.
    
    Please go to the {home} page or, if you think this is an error, to the {contact} page and contact me.",
        "portfolio.page.NotFoundPage.pageDescription": "Page not found.",
        "portfolio.page.NotFoundPage.pageTitle": "Not found : Johan Meester's portfolio",
        "portfolio.page.NotFoundPage.title": "NOT FOUND",
        "portfolio.page.SkillsPage.pageDescription": "Some of my skills.",
        "portfolio.page.SkillsPage.pageTitle": "My Skills : Johan Meester's portfolio",
        "portfolio.page.SkillsPage.title": "MY SKILLS",
        "portfolio.page.index.header.callToAction": "CONTACT ME",
        "portfolio.page.index.header.greeting": "WELCOME",
        "portfolio.page.index.header.intro": "I am {name}",
        "portfolio.page.index.header.look": "HAVE A LOOK AROUND",
        "portfolio.page.index.header.trade": "FRONT-END DEVELOPER",
        "portfolio.page.index.title": "Introduction",
        "portfolio.page.projects.title": "MY PROJECTS",
        "portfolio.social_links.header": "Also find me on:",
        "portfolio.title": "Welcome to my portfolio",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
  onSubmitHandler={[Function]}
/>
`;
